agent/doc/swdesign/README.md

```diff
@@ -161,6 +161,461 @@ Needs:
 - utest
 - itest
 
+#### AgentManager shall receive workload states of the workloads it manages
+`swdd~agent-manager-receives-workload-states-of-its-workloads~1`
+
+Status: approved
+
+The AgentManager shall receive the workload states of the workloads it manages.
+
+Rationale: The agent uses the workload states of its workloads for handling inter-workload dependencies of workloads on the same agent.
+
+Tags:
+- AgentManager
+
+Needs:
+- impl
+- utest
+
+#### AgentManager shall store the workload states of the workloads it manages
+`swdd~agent-stores-workload-states-of-its-workloads~1`
+
+Status: approved
+
+The AgentManager shall store the workload states of the workloads it manages into the ParameterStorage.
```

Comment:
Hint: I think, the name ParameterStorage will be changed to WorkloadStateStore in PR of @krucod3.
Maybe you could sync about the merge strategy.

Reply:
I will take care of it when merging the main branch.